internal/util/k8sutil/api.go

```diff
@@ -112,6 +112,9 @@ func GetCustomResourceDefinitions(crdsDir string) (
 func DefinitionsForV1CustomResourceDefinitions(crds ...apiextv1.CustomResourceDefinition) (keys []registry.DefinitionKey) {
 	for _, crd := range crds {
 		for _, ver := range crd.Spec.Versions {
+			if !ver.Served {
+				continue
```

Comment:
Should add a helpful debug log
```suggestion
				log.Debugf("Not adding unserved CRD %q version %q to set of owned keys", crd.GetName(), ver.Name)
				continue
```